src/core/data/ions/particle_initializers/maxwellian_particle_initializer.h

```diff
@@ -84,6 +83,24 @@ namespace core
 
 
     private:
+        // can be relocated if needed
+        static std::mt19937_64 getRNG(std::optional<size_t> const& seed)
+        {
+            if (!seed.has_value())
+            {
+                std::random_device randSeed;
+                std::seed_seq seed_seq{randSeed(), randSeed(), randSeed(), randSeed(),
+                                       randSeed(), randSeed(), randSeed(), randSeed()};
+                return std::mt19937_64(seed_seq);
+            }
+            return std::mt19937_64(*seed);
+        }
+
+        template <typename T = float>
+        static std::uniform_real_distribution<T> get_unif_dist(){
+          return std::uniform_real_distribution<T>(0, 1. - std::numeric_limits<T>::epsilon());
```

Comment:
1.- epsilon is a bit magical here.
I know what it's for but we should probably either put a comment to explain the context or return that from a properly named function?